pandas/tests/frame/methods/test_dtypes.py

```diff
@@ -135,13 +133,9 @@ def test_dtypes_timedeltas(self):
         )
         tm.assert_series_equal(result, expected)
 
-    @pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
     def test_frame_apply_np_array_return_type(self, using_infer_string):
         # GH 35517
         df = DataFrame([["foo"]])
         result = df.apply(lambda col: np.array("bar"))
-        if using_infer_string:
-            expected = Series([np.array(["bar"])])
-        else:
-            expected = Series(["bar"])
+        expected = Series(np.array(["bar"]), dtype=object)
```

Comment:
Hmm, shouldn't we expect str dtype here if that is enabled?

Reply:
I could see either way and I don't have a strong preference

Reply:
Ah, I missed that this is a kind of "reducing" apply, because the applied lambda returns an 0dim array (kind of an array scalar). 

When doing a normal apply preserving the column length, it already infers it as string:

```
In [25]: result = df.apply(lambda col: np.array(["bar"]))

In [26]: result
Out[26]: 
     0
0  bar

In [27]: result.dtypes
Out[27]: 
0    str
dtype: object
```

So here it is essentially reducing each column and then creating a Series with the results. Now, also in this case I would expect that we infer the dtype? 
But it seems this is not specific to strings, because also when doing the same with an integer, we get object dtype:

```
In [31]: result = df.apply(lambda col: np.array(1))

In [32]: result
Out[32]: 
0    1
dtype: object
```

Reply:
Ah, and the reason that it is object dtype is because we actually store the 0dim array object in the Series.. Continuing with the last example above:

```
In [33]: result.values
Out[33]: array([array(1)], dtype=object)
```

So yes, object dtype is correct here, but it's also just a strange test .. (I would say that ideally we "unpack" those 0dim arrays into actual scalars and then do proper type inference)

Reply:
Yea I'm not sure. I don't quite understand how this test is useful in practice, so hard to form an opinion

Reply:
The 0-dim numpy scalar in an object array is the expected result and is intentional, xref #46199

and the same as creating a Series with a scalar....

```python
>>> pd.Series(np.array("bar"))
0    bar
dtype: object
>>> pd.Series(np.array("bar")).item()
array('bar', dtype='<U3')
>>> 
```

so I think need to just be more explicit in the expected, i.e. `expected = Series(np.array("bar"))` and this will also pass with both `future.infer_string = True` and `future.infer_string = False`

I think `Series(np.array("bar"))` is more explicit than `Series(np.array(["bar"]), dtype=object)` even though they both compare equal in testing...

```python
>>> pd.Series(np.array(["bar"]), dtype=object)
0    bar
dtype: object
>>> pd.Series(np.array(["bar"]), dtype=object).item()
'bar'
>>> import pandas._testing as tm
>>> tm.assert_series_equal(pd.Series(np.array("bar")), pd.Series(np.array(["bar"]), dtype=object))
>>> 
```